### DISCUSSION TITLE
Make docker startup foolproof regarding mapped volumes' permissions

### DISCUSSION THREAD

**Discussion by t7tran** [directus/directus#7111] [Feature Requests]
Since Docker user changed from `root` in rc.71, it has caused confusion to some users - running Directus on both Docker and Kubernetes. Let's add some diagnoses into the start up process so it would show errors and exit the process.

This kind of logic would be best to be placed in an entrypoint script which has recently been removed. So please comment how this would play out.

Regarding of the checks, what should be included?

- Check read permission for `./database` folder
- Check read permission for `./extendsions` folder
- Check read-write permission for `./uploads` folder
- Apply to all above, do we need to run the check recursively to all descendant files/folders?

**Comment by rijkvanzanten** [comment]
I think the most important thing is providing the necessary information. Right now, people are able to replace the data folder, without Directus/Docker warning them about the implications. Only when they start doing something that's illegal, like uploading a file, do things start to crash. This in turn makes it feel like Directus itself is broken, causing duplicate issue reports.

Would it be possible to do a check on startup to see if the folder exists and has the right permissions? 
Docker itself doesn't have any scripting capabilities during startup right? And we don't have bash installed on the image? 

Alternatively, we could do a check in Directus itself on startup to see if all the connected services are healthy and available 🤔

**Reply by paescuj** [reply]
I suggest to add a little check in Directus itself, e.g. right here:
https://github.com/directus/directus/blob/main/api/src/app.ts#L48-L65

Adding a check in Directus itself means:
* Docker users as well as users who have installed Directus manually could both profit from this check.
* No need to add any logic in the Docker image (no bash, no entrypoint script, ...).

Also I think we should just check the permission of the root folders itself (e.g. using [fsPromises.access](https://nodejs.org/api/fs.html#fs_fspromises_access_path_mode)) and not recursively. Checking all descendant files/folders could mean a lot of work and would slow up the startup time.

In my opinion, we should just print a warning without exiting Directus - some users may deliberately want stricter permissions.

**Reply by rijkvanzanten** [reply]
Makes sense to me! A `validateStorage` function would make total sense in that context! Wanna open a PR for that?

**Reply by paescuj** [reply]
> Makes sense to me! A `validateStorage` function would make total sense in that context! Wanna open a PR for that?

Will do!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
